buildtools/cli.go

```diff
@@ -822,23 +801,42 @@ func NpmInstallCmd(c *cli.Context) error {
 	if show, err := cliutils.ShowGenericCmdHelpIfNeeded(c, c.Args(), "npminstallhelp"); show || err != nil {
 		return err
 	}
-	return npmInstallCiCmd(c, npm.NewNpmInstallCommand())
+	return npmGenericCmd(c, "install", true)
 }
 
 func NpmCiCmd(c *cli.Context) error {
 	if show, err := cliutils.ShowGenericCmdHelpIfNeeded(c, c.Args(), "npmcihelp"); show || err != nil {
 		return err
 	}
-	return npmInstallCiCmd(c, npm.NewNpmCiCommand())
+	return npmGenericCmd(c, "ci", true)
 }
 
-func npmInstallCiCmd(c *cli.Context, npmCmd *npm.NpmInstallOrCiCommand) error {
+func npmGenericCmd(c *cli.Context, cmdName string, collectBuildInfo bool) error {
+	if show, err := cliutils.ShowCmdHelpIfNeeded(c, c.Args()); show || err != nil {
+		return err
+	}
+	orgArgs := c.Args()
+	//cmdName, _ := getCommandName(orgArgs)
+	switch cmdName {
+	// Aliases accepted by npm.
+	case "i", "isntall", "add", "install":
+		cmdName = "install"
+		collectBuildInfo = true
+	case "ci":
+		collectBuildInfo = true
+	case "publish", "p":
+		return NpmPublishCmd(c)
+	}
+
+	// Run generic npm command.
+	npmCmd := npm.NewNpmCommand(cmdName, collectBuildInfo)
 	configFilePath, args, err := GetNpmConfigAndArgs(c)
 	if err != nil {
 		return err
 	}
-
-	npmCmd.SetConfigFilePath(configFilePath).SetArgs(args)
+	npmCmd.SetConfigFilePath(configFilePath)
+	npmCmd.CommonArgs.SetNpmArgs(orgArgs)
+	npmCmd.CommonArgs.SetNpmArgs(args)
 	err = npmCmd.Init()
 	if err != nil {
 		return err
```

Comment:
Can be simplified to - 
```
	if err = npmCmd.Init(); err != nil {
		return err
	}
```
The same goes for the code in the `NpmPublishCmd` function.

Reply:
Done